how-it-works/1_protocol/02_consensus.subpage.md

```diff
@@ -5,14 +5,118 @@ shareImage: /img/how-it-works/consensus.600.jpg
+In this case, the resource consumption (networking bandwidth, CPU) and time to create the block for the round increases as multiple block proposals are created, broadcast, and notarized by nodes.
+The IC protocol ensures that the subnet performance degrades gracefully with increasing faults in the subnet communication or of nodes.
+
+[](/img/how-it-works/consensus_notarization.png)
+Notarization of increasing-rank block proposals in a round
+
+## Finalization
+
+Given the above-described scheme that allows up to all the nodes of a subnet to create and propose a block for a given round if things go wrong, more than one block can become fully notarized in a round.
+In this case it is not clear, which of the blocks should be the chosen one for this round.
+That means that we need an additional mechanism for *determining and observing finality*, i.e., which of the possibly multiple fully-notarized blocks of the round should be the one that is the chosen (finalized) block for the round which can be provided to the upper layers of the protocol stack to be executed.
+
+It is far from reality to assume a synchronous communication model in practice
+This means that we cannot assume to be able to deduce finality by waiting for a timeout to occur and then conclude that no further notarization shares can be received for proposed blocks.
+We must work under the assumption of an asynchronous communication model where network messages may be arbitrarily delayed, but will eventually reach all nodes of a subnet.
+This model reflects reality of communication networks, but also makes it much harder to determine finality.
+
+The option of choosing "probabilistic finality," similar to what is done in Bitcoin-like protocols, by considering a block final once a sufficient number of blocks have built on top of it in the blockchain is not acceptable for the IC for two reaons: (1) Probabilistic finality is only a very weak notion of finality and (2) probabilistic finality would increase the time to finality drastically.
+
+Considering these unacceptable approaches to finality, the IC uses a cryptographic protocol for determining finality.
+This protocol uses, like the notarization protocol, BLS multi signatures to determine finality:
+A node that observes a fully notarized block issues a finalization signature share (finalization share) on the block and broadcasts it to the nodes of the subnet, but does so *only if it has not notary signed another block in this round*.
+The latter condition guarantees that if there is a block that receives finality shares from at least two thirds of the nodes, then there can only be one such block.
+That is, if there is a fully finalized block in a given round, then this is a proof that this block is the only finalized block of this round.
```

Comment:
You say: "this block is the only finalized block". Does you mean "fully finalized" here? I think so.

One could also say: "this is the only fully notarized block", which is a stronger statement, and I think we even need this property in the full analysis of the protocol